client/js/js_utils.js

```diff
@@ -26,181 +26,97 @@ function addLoadEvent(func) {
-  }
-  else {
-    // code for IE6, IE5
-    xmlhttp = new ActiveXObject('Microsoft.XMLHTTP');
-  }
 
-  xmlhttp.onreadystatechange = function() {
-    if (xmlhttp.readyState == XMLHttpRequest.DONE) {
-      if (xmlhttp.status == 201) {
-        on_success_callback(xmlhttp.responseText);
-      }
-      else if (xmlhttp.status == 404) {
-        try {
-          on_failure_callback();
-        }
-          catch (err) {
-          on_failure_callback(xmlhttp.responseText);
-        }
-      }
-      else {
-        try {
-          on_failure_callback();
-        }
-        catch (err) {
-          on_failure_callback(xmlhttp.responseText);
-        }
-      }
+  xhr.onload = function(e) {
+    if (200 == this.status) {
+      on_success_callback(this.response);
+    }
+    else {
+      on_failure_callback(this.response);
     }
   };
-  xmlhttp.open('POST', url, true);
-  xmlhttp.setRequestHeader('Fiware-Service', 'tourguide');
-  xmlhttp.setRequestHeader('Content-type', 'application/json');
-  xmlhttp.send(JSON.stringify(data));
+
+  xhr.onerror = function(e) {
+      on_failure_callback(this.response);
+  };
+
+
+  xhr.open('POST', url, true);
+  xhr.setRequestHeader('Fiware-Service', 'tourguide');
+  xhr.setRequestHeader('Content-type', 'application/json');
+  xhr.send(JSON.stringify(data));
 }
 
 
 function patch_ajax_petition(url, on_success_callback, on_failure_callback, data) {
-  console.log('patch debug:');
-  console.log('url: ' + url);
-  console.log(data);
-  var xmlhttp;
-
-  if (window.XMLHttpRequest) {
-    // code for IE7+, Firefox, Chrome, Opera, Safari
-    xmlhttp = new XMLHttpRequest();
-  }
-  else {
-    // code for IE6, IE5
-    xmlhttp = new ActiveXObject('Microsoft.XMLHTTP');
-  }
 
-  xmlhttp.onreadystatechange = function() {
-    if (xmlhttp.readyState == XMLHttpRequest.DONE) {
-      if ((xmlhttp.status == 201) || xmlhttp.status == 204) {
-        on_success_callback(xmlhttp.responseText);
-      }
-      else if (xmlhttp.status == 404) {
-        try
-        {
-          on_failure_callback();
-        }
-        catch (err)
-        {
-          on_failure_callback(xmlhttp.responseText);
-        }
-      }
-      else {
-        try {
-          on_failure_callback();
-        }
-        catch (err) {
-          on_failure_callback(xmlhttp.responseText);
-        }
-     }
+  var xhr = new XMLHttpRequest();
+
+  xhr.onload = function(e) {
+    if ((201 == this.status) || (204 == this.status)) {
+      on_success_callback(this.response);
+    }
+    else {
+      on_failure_callback(this.response);
     }
   };
-  xmlhttp.open('PATCH', url, true);
-  xmlhttp.setRequestHeader('Fiware-Service', 'tourguide');
-  xmlhttp.setRequestHeader('Content-type', 'application/json');
-  xmlhttp.send(JSON.stringify(data));
-}
 
+  xhr.onerror = function(e) {
+      on_failure_callback(this.response);
+  };
+  xhr.open('PATCH', url, true);
+  xhr.setRequestHeader('Fiware-Service', 'tourguide');
```

Comment:
this code is repeated all over many places.

Why don't you create an xhr builder a centralize this process in one place